common/date.go

```diff
@@ -24,6 +24,27 @@ func (d Date) MarshalJSON() ([]byte, error) {
 	return []byte(d.quote()), nil
 }
 
+// UnmarshalText Parses the text string to the Date
+func (d *Date) UnmarshalText(text []byte) (err error) {
+	s := strings.Trim(string(text), `"`)
+	parsedTime, err := time.Parse(layout, s)
+	*d = Date(parsedTime)
+	return
+}
+
+// MarshalText marshall Date into Text
+func (d Date) MarshalText() ([]byte, error) {
+	return []byte(d.quote()), nil
+}
+
+// UnmarshalParam Parses the form's value to the Date
+func (d *Date) UnmarshalParam(param string) (err error) {
```

Comment:
Need to implement their `BindUnmarshaler` interface 
https://github.com/gin-gonic/gin/pull/3933